core/network/src/custom_proto/behaviour.rs

```diff
@@ -930,6 +930,7 @@ where
 			CustomProtoHandlerOut::ProtocolError { error, .. } => {
 				debug!(target: "sub-libp2p", "Handler({:?}) => Severe protocol error: {:?}",
 					source, error);
+				self.peerset.report_peer(source.clone(), i32::min_value());
```

Comment:
Maybe add a comment here explaining the common situation for `ProtocolError`. What you have in the PR description is good enough.